pkg/operator/defragcontroller/defragcontroller.go

```diff
@@ -186,6 +181,39 @@ func (c *DefragController) checkDefrag(ctx context.Context, recorder events.Reco
 	return v1helpers.NewMultiLineAggregate(errors)
 }
 
+func (c *DefragController) ensureControllerDisabledCondition(recorder events.Recorder) error {
+	controlPlaneTopology, err := ceohelpers.GetControlPlaneTopology(c.infrastructureLister)
+	if err != nil {
+		return fmt.Errorf("failed to get control-plane topology: %w", err)
+	}
+
+	// Defrag is blocking and can only be safely performed in HighlyAvailableTopologyMode
+	if controlPlaneTopology == configv1.HighlyAvailableTopologyMode {
+		_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient,
+			v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+				Type:   defragDisabledCondition,
+				Status: operatorv1.ConditionFalse,
+				Reason: "AsExpected",
```

Comment:
I forget how come we always do reason "AsExpected", I believe I might have asked this before sorry :)